serverlessworkflow/modules/ROOT/pages/cloud/operator/using-persistence.adoc

```diff
@@ -29,82 +36,261 @@ spec:
+
+The SonataFlow CR facilitates the configuration of the persistence with workflow scope, and you can use it independently if the SonataFlowPlatform persistence was already configured in the current namespace, see: <<_persistence_configuration_precedence_rules, Persistence configuration precedence rules>>.
+
+To configure the persistence, you must use the `persistence` field in the SonataFlow CR `spec`:
+
+.SonataFlow CR persistence configuration example
+[source,yaml]
+----
 apiVersion: sonataflow.org/v1alpha08
 kind: SonataFlow
 metadata:
-  name: callbackstatetimeouts
+  name: example-workflow
   annotations:
-    sonataflow.org/description: Callback State Timeouts Example k8s
+    sonataflow.org/description: Example Workflow
     sonataflow.org/version: 0.0.1
 spec:
-  persistence: {}
-  ...
----
+  persistence:
+    postgresql:
+      secretRef:
+        name: postgres-secrets <1>
+        userKey: POSTGRESQL_USER <2>
+        passwordKey: POSTGRESQL_PASSWORD <3>
+      serviceRef:
+        name: postgres <4>
+        namespace: sonataflow-infra <5>
+        databaseName: sonataflow <6>
+        databaseSchema: my-example-schema <7>
+        port: 1234 <8>
+  flow:
+     ...
+----
+
+<1> Name of the link:{k8n_secrets_url}[Kubernetes Secret] containing the username and password to connect with the database.
+<2> Name of the link:{k8n_secrets_url}[Kubernetes Secret] `key` containing the username to connect with the database.
+<3> Name of the link:{k8n_secrets_url}[Kubernetes Secret] `key` containing the password to connect with the database.
+<4> Name of the Kubernetes Service to connect with the PostgreSQL database server.
+<5> (Optional) Kubernetes namespace containing the PostgreSQL Service. Defaults to the workflow's local namespace.
+<6> Name of the PostgreSQL database to store the workflow's data.
+<7> (Optional) Name of the database schema to store workflow's data. Defaults to the workflow's name.
+<8> (Optional) Port number to connect with the PostgreSQL Service. Defaults to 5432.
 
-This configuration signals the operator that the workflow requires persistence and that it expects its configuration to be populated accordingly. 
-The operator will add the relevant JDBC properties in the `application.properties`
-generated and as part of the pod´s environment so that it can connect to the persistence service defined in the `Platform` CR.
+This configuration signals the operator that the current workflow must be properly configured to connect with that PostgreSQL database server when deployed.
+Similar to the SonataFlowPlatform persistence, the operator will add the relevant JDBC connection parameters in the form of <<persistence_env_vars_config_example, environment variables>> to the workflow container.
+
+Additionally, for SonataFlow CR deployments that use the `preview` profile, it can configure the {product_name} build system to include specific Quarkus extensions required for persistence.
```

Comment:
```suggestion
Additionally, for `SonataFlow` CR deployments that use the `preview` profile, it can configure the {product_name} build system to include specific Quarkus extensions required for persistence.
```